lib/Parser/parse.cpp

```diff
@@ -5445,6 +5456,11 @@ void Parser::ParseFncFormals(ParseNodePtr pnodeFnc, ushort flags)
                 }
             }
 
+            if ((IsStrictMode() || isNonSimpleParameterList) && m_currentScope->GetHasDuplicateFormals())
```

Comment:
We already had this error checked for in strict mode but I don't see that you've removed it here so is it not redundant to have `IsStrictMode()` in the condition here?